### PR TITLE
USDC faucet for base sepolia 

### PR DESCRIPTION
**Pull request by jazz-cb**

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: src/coinbase/address.ts [hunk 1]
```diff
@@ -255,14 +255,20 @@ export class Address {
    * Requests faucet funds for the address.
    * Only supported on testnet networks.
    *
+   * @param asset_id - The ID of the asset to transfer from the faucet.
    * @returns The faucet transaction object.
    * @throws {InternalError} If the request does not return a transaction hash.
    * @throws {Error} If the request fails.
    */
-  public async faucet(): Promise<FaucetTransaction> {
+  public async faucet(asset_id?: string): Promise<FaucetTransaction> {
```
Comment (John-peterson-coinbase) [inline src/coinbase/address.ts:263]:
```suggestion
  public async faucet(assetId?: string): Promise<FaucetTransaction> {
```

File: src/coinbase/address.ts [hunk 2]
```diff
@@ -255,14 +255,20 @@ export class Address {
...
+    if (!asset_id) {
+      asset_id = "";
+    }
```
Comment (John-peterson-coinbase) [inline src/coinbase/address.ts:266]: Is this necessary? 

`requestExternalFaucetFunds` takes an optional `assetId?`. Do we need it to be explicitly set to `""`?

File: src/coinbase/address.ts (hunk 1 above)
Comment (jazz-cb) [reply src/coinbase/address.ts]: fixed

File: src/coinbase/address.ts (hunk 2 above)
Comment (jazz-cb) [reply src/coinbase/address.ts]: removed

**Review by alex-stone — APPROVED** [review]

**Pull request merged**